exercises/binary-search/example.cpp

```diff
@@ -0,0 +1,40 @@
+#include <stdexcept>
+#include "binary_search.h"
+
+namespace binary_search {
+
+size_t find(std::vector<int> const& data, int const value)
+{
+    if(data.empty() || value < data.front() || data.back() < value)    
+    {
+        throw std::domain_error("Value not in input vector!");
+    }
+    
+    size_t lpos {0};
+    size_t rpos {data.size() - 1};
+    size_t mid  {0};
+
+    while (lpos <= rpos)
+    {
+        mid = (lpos + rpos) >> 1;
```

Comment:
Technically this doesn't fix the issue. You need
```
mid = lpos + ((rpos - lpos) / 2);
```
The fix in the blog post that they cite for Cpp was using signed indices (lpos and rpos), so they could just cast them to unsigned and the addition was guaranteed to never overflow due to the extra bit. But that's assuming the indices even fit into a signed int.

(Since you did want more reading, we do use size_t here for indexes, but there is a good argument for using a signed type -> isocpp/CppCoreGuidelines#1115. )

Reply:
Oh, I see. I changed the LOC accordingly.

NB: Thanks for the link, very interesting read! I understand we'd run into a narrowing conversion issue if we were to change size_t for ptrdiff_t, since vector.size() returns an unsigned type. Hope that the index type will trickle down into the standard not too far in the future...